### PR TITLE
Issue 5264 - Ticket filters: string filter form

### LINKED ISSUE #5264: Ticket filters: string filter form
### Description
Product issue: https://github.com/3drepo/3D-Repo-Product-Team/issues/521
Properties depicted as string will use this filter form. Also the basic fields for tickets which are not defined as "property"; title, description and id.

String properties have the following criterias  for filtering 
image

When selecting 'equals', 'does not equals', 'contains', 'does not contains'  a text field will appear with a + symbol will appear to the right of the field. Clicking on + will create another field so the user can search any string that matches several strings. If there is more than one field  - will be enabled to let the user get rid of fields if they change their mind.
image
Once the user filled these fields apply will be enabled and if clicked here it will add the filter to the list.

If the user selected 'Exists' or 'Does not exists' the no other field will be presented and the apply button will be enabled.

### Goals
- [ ] As a user I want to filter by strings in the tickets list

### Tasks
- [ ] Create filter form specific to string (operators and input type)
- [ ] Add string validation
- [ ] The structure is very similar to group filters, see if that can be re-used

### Related Resources
[MockUp](https://www.figma.com/design/nboKoEopFrXIrOAYTJQHru/Filters?node-id=823-43606&node-type=section&t=5P2O7r9RqRYKLdbp-0)

### PR DESCRIPTION
**Pull request by Amantini1997**
This fixes #5264

#### Description
Added string filter handling:
- show only valid operators;
- allow fields count and their addition/removal depending on the operator
- validate form for string

#### Test cases
Test the different operators:
- [not ]exists should be always valid;
- other operators should have the fields filled to be valid. Furthermore, there's a limit on the string depending on if it is a text (120 chars) or a long text (1,200);
- adding and removing the fields should work as expected;
- the form should handle any fields changes (if needed) upon changing the operator

### CODE REVIEW

File: frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/ticketGroups/groups/groupRulesForm/groupRulesInputs/ruleFieldValues/ruleFieldValues.component.tsx [hunk 1]
```diff
@@ -56,15 +54,19 @@ export const RuleFieldValues = ({ disabled }) => {
...
-						</ValueIconContainer>
-					</ValuesContainer>
+					<ArrayFieldContainer
+						key={field.id}
+						onRemove={() => remove(i)}
+						disableRemove={disabled || fields.length === 1}
+						onAdd={appendEmptyValue}
+						disableAdd={disabled || i !== (fields.length - 1)}
+					>
+						<FieldValueInput
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/ticketGroups/groups/groupRulesForm/groupRulesInputs/ruleFieldValues/ruleFieldValues.component.tsx:64]: image

File: frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/ticketGroups/groups/groupRulesForm/groupRulesForm.helpers.ts [hunk 2]
```diff
@@ -52,18 +53,18 @@ export const formRuleToGroupRule = ({ values, field, ...rule }: IFormRule): IGro
...
-	values: (values || []).map((v) => v.value),
+	values: mapFormArrayToArray(values || []),
 });
 
 export const groupRuleToFormRule = ({ values, field, ...rule }: IGroupRule): IFormRule => ({
 	...rule,
 	field: {
 		operator: field.operator,
-		values: (field.values || []).map((value) => ({ value })),
+		values: mapArrayToFormArray(field.values || []),
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/ticketGroups/groups/groupRulesForm/groupRulesForm.helpers.ts:65]: maybe `mapArrayToFormArray` can cater for undefined or null values and treat them as []. You  reckon there could be a problem with that?

File: frontend/src/v5/ui/controls/inputs/arrayFieldContainer/arrayFieldContainer.component.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import { IconContainer, Container } from './arrayFieldContainer.styles';
+
+type ArrayFieldContainerProps = {
+	children: any,
+	disableRemove?: boolean,
+	disableAdd?: boolean,
+	onRemove: () => void,
+	onAdd: () => void,
+};
+export const ArrayFieldContainer = ({ children, onRemove, onAdd, disableAdd, disableRemove }: ArrayFieldContainerProps) => (
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/controls/inputs/arrayFieldContainer/arrayFieldContainer.component.tsx:29]: Can we add a couple of  examples of this in storybook? maybe with a textfield and another one with a select as children.

**Review by sanmont3drepo — CHANGES_REQUESTED** [review]

File: frontend/src/v5/ui/components/viewer/cards/cardFilters/filterForm/filterFormValues/filterFormValues.componen.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
+/**
```
[the remark this answers is not in the document]
Comment (Amantini1997) [reply frontend/src/v5/ui/components/viewer/cards/cardFilters/filterForm/filterFormValues/filterFormValues.componen.tsx:1]: Dan, is that you?

**Review by sanmont3drepo — CHANGES_REQUESTED** [review]

**Review by sanmont3drepo — APPROVED** [review]

**Review by The-Daniel — CHANGES_REQUESTED** [review]

**Review by The-Daniel — APPROVED** [review]

**Pull request merged**